cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -53,9 +54,88 @@ const (
 )
 
 type cliOptions struct {
-	sharedDevNum       int
-	enableMonitoring   bool
-	resourceManagement bool
+	preferredAllocationPolicy string
+	sharedDevNum              int
+	enableMonitoring          bool
+	resourceManagement        bool
+}
+
+type preferredAllocationPolicyFunc func(*pluginapi.ContainerPreferredAllocationRequest) []string
+
+// nonePolicy is used for allocating GPU devices randomly.
+func nonePolicy(req *pluginapi.ContainerPreferredAllocationRequest) []string {
+	klog.V(2).Info("Select nonePolicy for GPU device allocation")
+
+	deviceIds := req.AvailableDeviceIDs[0:req.AllocationSize]
```

Comment:
This `req.AllocationSize` comes from an external process. I think a static analyzer (e.g. checkmarx) may consider this input data non-trusted and problematic. I'd recommend to clamp down the value to what ever is less, i.e.
`deviceIds := req.AvailableDeviceIDs[:min(req.AllocationSize, len(req.AvailableDeviceIDs))]`

Reply:
In kubelet, **func devicesToAllocate**
`if needed < aligned.Len()  `
`               preferred, err := m.callGetPreferredAllocationIfAvailable(...)`

I think here has already ensured that the `req.AllocationSize` is less than `len(req.AvailableDeviceIDs)`.